agent/sighandlers/termination_handler.go

```diff
@@ -61,11 +66,11 @@ func FinalSave(saver statemanager.Saver, taskEngine engine.TaskEngine) error {
 	engineDisabled := make(chan error)
 
 	disableTimer := time.AfterFunc(engineDisableTimeout, func() {
-		engineDisabled <- errors.New("Timed out waiting for TaskEngine to settle")
+		engineDisabled <- errors.New("final save: timed out waiting for TaskEngine to settle")
```

Comment:
nit: **F**inal save:...

Reply:
This is conventional idiomatic go casing for the error message to be lower cased. Here's an example:

`agent error: this package errored fetching data: I couldn't reach the server.`

This can come from 3 different sources and would look weird if it was capitalized after each `:`.

Reply:
The convention is described [here](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).